### DISCUSSION TITLE
feat: support custom inline attributes

### DISCUSSION THREAD

**Discussion by mrverdant13** [singerdmx/flutter-quill#584] [Feature Request] [answered]
## Details

Expose a way to register custom inline attributes.

Devs often need a way to use custom attributes depending on some specific use cases. i.e. mentions, tags, etc.

## Related issues and discussions

- Mentions: https://github.com/singerdmx/flutter-quill/discussions/564
- Custom image: https://github.com/singerdmx/flutter-quill/discussions/559
- Custom embed parameters: https://github.com/singerdmx/flutter-quill/discussions/569

I am working on a really simple PR that might enable this feature but would like to have additional feedback once I submit it.

**Comment by singerdmx** [comment]
No I am against the idea

**Comment by singerdmx** [comment]
Anything needs to be added should be directly added into repo

**Comment by singerdmx** [comment]
Such PR will not be approved

**Comment by singerdmx** [comment]
Custom styling is already supported 
No custom attributes 
If you need any new attributes, it should be in the repo

**Comment by mrverdant13** [comment]
That's a shame.
This would promote the development of external packages to be plugged to `flutter_quill`.
No worries.

**Comment by singerdmx** [comment]
We have enough bugs to fix, e.g. https://github.com/singerdmx/flutter-quill/issues/561 (I think you are working on this).
We don't need unnecessary complication to the repo. This will only make maintainability worse.
Also we don't want people just to use this repo without contributing to it such as adding more inline attribute feature.

**Comment by singerdmx** [comment, marked as answer]
All inline attributes supported should be explicitly whitelisted here in this repo.
